### PR TITLE
DOCS-4056 Adds CodePush Documentation

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do?

Single sources a doc page for Appcenter Codepush into the React Native Monitoring docs.

### Motivation

DOCS-4056

https://docs-staging.datadoghq.com/alai97/reactnative-codepush-doc-update/real_user_monitoring/reactnative/codepush

### Additional Notes

Remember to update the `pull_config.yaml` file back to `main` before merging!

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by hestonhoffman — COMMENTED** [review]
I can't hit `/real_user_monitoring/reactnative/codepush/` on a local build 🤔

**Review by hestonhoffman — APPROVED** [review]
Looks good! I'll let you merge :)

File: local/bin/py/build/configurations/pull_config.yaml [hunk 1]
```diff
@@ -806,6 +806,25 @@
             - link: "real_user_monitoring/explorer/"
               tag: "Documentation"
               text: "Learn how to explore your RUM data"
+    - action: pull-and-push-file
+      branch: develop
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config.yaml:810]:
```suggestion
      branch: main
```

File: local/bin/py/build/configurations/pull_config.yaml [hunk 2]
```diff
@@ -806,6 +806,25 @@
...
+      globs:
+      - 'docs/codepush.md'
+      options:
+        dest_path: '/real_user_monitoring/reactnative/'
+        file_name: 'codepush.md'
+        front_matters:
+          dependencies: ["https://github.com/DataDog/dd-sdk-reactnative/blob/develop/docs/codepush.md"]
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config.yaml:817]:
```suggestion
          dependencies: ["https://github.com/DataDog/dd-sdk-reactnative/blob/main/docs/codepush.md"]
```

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 3]
```diff
@@ -807,6 +807,27 @@
             - link: "real_user_monitoring/explorer/"
               tag: "Documentation"
               text: "Learn how to explore your RUM data"
+
+    - action: pull-and-push-file
+      branch: develop
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config_preview.yaml:812]:
```suggestion
      branch: main
```

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 4]
```diff
@@ -807,6 +807,27 @@
...
+      globs:
+      - 'docs/codepush.md'
+      options:
+        dest_path: '/real_user_monitoring/reactnative/'
+        file_name: 'codepush.md'
+        front_matters:
+          dependencies: ["https://github.com/DataDog/dd-sdk-reactnative/blob/develop/docs/codepush.md"]
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config_preview.yaml:819]:
```suggestion
          dependencies: ["https://github.com/DataDog/dd-sdk-reactnative/blob/main/docs/codepush.md"]
```

**Pull request merged**